hydra-cluster/test/Test/Hydra/Cluster/FaucetSpec.hs

```diff
@@ -30,27 +30,29 @@ spec = do
               -- 10 unique outputs
               length (fold utxos) `shouldBe` 10
   describe "returnFundsToFaucet" $
-    it "seedFromFaucet and returnFundsToFaucet should work together" $ do
-      showLogsOnFailure "FaucetSpec" $ \tracer ->
-        withTempDir "hydra-cluster" $ \tmpDir ->
-          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
-            let faucetTracer = contramap FromFaucet tracer
-            actor <- generate $ elements [Alice, Bob, Carol]
-            (vk, _) <- keysFor actor
-            (faucetVk, _) <- keysFor Faucet
-            initialFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            seeded <- seedFromFaucet node vk 100_000_000 faucetTracer
-            returnFundsToFaucet faucetTracer node actor
-            remaining <- queryUTxOFor networkId nodeSocket QueryTip vk
-            finalFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
-            foldMap txOutValue remaining `shouldBe` mempty
+    prop "seedFromFaucet and returnFundsToFaucet should work together" $
+      withMaxSuccess 10 $
+        forAll (Coin <$> choose (1000000, 10000000000)) $ \coin -> do
+          showLogsOnFailure "FaucetSpec" $ \tracer ->
+            withTempDir "hydra-cluster" $ \tmpDir ->
+              withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \node@RunningNode{networkId, nodeSocket} -> do
+                let faucetTracer = contramap FromFaucet tracer
+                actor <- generate $ elements [Alice, Bob, Carol]
+                (vk, _) <- keysFor actor
+                (faucetVk, _) <- keysFor Faucet
+                initialFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
+                seeded <- seedFromFaucet node vk coin faucetTracer
+                returnFundsToFaucet faucetTracer node actor
+                remaining <- queryUTxOFor networkId nodeSocket QueryTip vk
+                finalFaucetFunds <- queryUTxOFor networkId nodeSocket QueryTip faucetVk
+                foldMap txOutValue remaining `shouldBe` mempty
 
-            -- check the faucet has one utxo extra in the end
-            length seeded `shouldBe` length remaining + 1
+                -- check the faucet has one utxo extra in the end
+                length seeded `shouldBe` length remaining + 1
```

Comment:
I don't understand this assertion. (I know you have not changed it, but still).

I think the equation is not wrong here, but it does not really match the comment above? It was introduced in https://github.com/input-output-hk/hydra/commit/0cfc4e0db9b2dd3ed0113af7c3274f4fb2166bec

Checking that the faucet has one utxo more than before would be based on `initialFaucetFunds` and `finalFaucetFunds`:

```suggestion
                length finalFaucetFunds `shouldBe` length initialFaucetFunds + 1
```